src/Booking.js

```diff
@@ -0,0 +1,26 @@
+class Booking {
+  constructor(bookingData) {
+    this.id = bookingData.id;
+    this.userID = bookingData.userID;
+    this.date = bookingData.date;
+    this.roomNumber = bookingData.roomNumber;
+    this.roomServiceCharges = bookingData.roomServiceCharges;
+    this.roomsArr = [];
+  }
+
+  // unavailableRooms(date, roomDate, bookingDate){
+  //   let result = 
+
+  // }
+
+  availableRooms(date, bookingDate){
```

Comment:
Do you want this to be inside of your booking class or inside of User? Wouldn't User look at the whole booking array? Just a question to ponder!